blockchain/headerchain.go

```diff
@@ -407,8 +407,15 @@ func (hc *HeaderChain) SetHead(head uint64, updateFn UpdateHeadBlocksCallback, d
 		//
 		// Update head first(head fast block, head full block) before deleting the data.
 		if updateFn != nil {
-			if err := updateFn(parent); err != nil {
-				return err
+			latestBlkNum, err := updateFn(parent)
+			if err != nil {
+				// rewound blockchain to genesis block
+				head = 0
```

Comment:
I'm not sure if this is ok. The previous logic stops at error, while the new logic resets to genesis at error. Could you tell me why this logic is required?

Reply:
My apologies, it should return an error. No need to delete anything further at this point. Updated.